pkg/common/hostname.go

```diff
@@ -0,0 +1,83 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// HostnameStore -
+type HostnameStore interface {
```

Comment:
@abays hey, I ninja'd in your function here and modified it to use this interface for hostnames. See the ControllerVM object above where I implemented the interface. This way both BaremetalSet and ControllerVM can share it but have different Status's to store the results.

Reply:
Cool!  This abstraction is a good idea